### ISSUE TITLE
Implied inheritance for contacts/contact_groups does not work when service objects have contacts or contact_groups defined

### ISSUE THREAD

**Issue by box293** [NagiosEnterprises/nagioscore#87]
I don't think this is a bug however I do think it should be raised so a discussion can be started on this topic.

I have a host and service
Host has contacts and contact_groups directives defined
Service has NO contacts or contact_groups directives defined
objects.cache shows the contacts and contact_groups applied to the host object
objects.cache shows the contacts and contact_groups applied to the service object, hence inheritance works

I go and add a contact to the service object (no contactgroup defined on the service object, only a contact)
Restart Nagios
objects.cache shows the contacts and contact_groups applied to the host object (nothing changed)
objects.cache no longer has the contact_groups object defined for the service object (inheritance NOT working) however the contacts directive is there

Why is the contact_groups not being inherited? contacts and contact_groups are two separate directives.

I did read the Object Inheritance documentation here https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/4/en/objectinheritance.html however it isn't very clear if what I am trying to do is possible.

ON a different but similar issue.
The host object has nagiosadmin as the contacts
The contacts I added on the service is another_contact with a +
however objects.cache only shows another_contact instead of nagiosadmin,another_contact

Here are some configs so you can play with the examples I mentioned above:
```
define contact{
        contact_name                    test_contact
        use                             generic-contact
        alias                           Test Contact
        email                           test@test.local
        }

define contact{
        contact_name                    another_contact
        use                             generic-contact
        alias                           Another Contact
        email                           another@test.local
        }

define contactgroup {
	contactgroup_name             	test_contactgroup
	alias                         	test_contactgroup
	members                       	test_contact
	}	

define host {
        host_name                       test_host
        alias                           test_host
        address                         127.0.0.1
        check_command                   check-host-alive!!!!!!!!
        initial_state                   o
        max_check_attempts              5
        check_interval                  5
        retry_interval                  1
        active_checks_enabled           1
        check_period                    24x7
        contacts                        nagiosadmin
        contact_groups                  test_contactgroup
        notification_period             24x7
        register                        1
        }

define service {
        host_name                       test_host
        service_description             Testing Stuff
        check_command                   check-host-alive!!!!!!!!
        max_check_attempts              5
        check_interval                  5
        retry_interval                  1
        check_period                    24x7
        notification_period             24x7
        contacts                        another_contact
        register                        1
        }
```

**Comment by giteric** [comment]
Regarding the first half, I think it is a bug . Only "contact_groups" are specified as being implied inherited values.  Not "contacts."  (According to the doc.)  So if you're setting contacts but not contact_groups, I would expect to see contact_groups inherited and contacts set with what you're setting.  It looks like the parser is abandoning inheritance of the contact_groups when it sees a valid contacts directive, even though the documentation indicates that it should still inherit the value.

In fact, the documentation implies that it should NOT be inheriting the contacts directive at all, so there may be a problem with the documentation AND a problem with the code.  :-)

**Comment by hedenface** [comment]
I agree that this is a bug. Pending further review, I'm setting it for 4.3.3

**Comment by nook24** [comment]
The commit says this is the wanted behavior: https://github.com/NagiosEnterprises/nagioscore/commit/01621635b394018ba9a5cd439eca407549f0319d

**Comment by tsadpbb** [comment]
I agree this is weird behavior and I would want it changed. Why remove the ability to inherit a contact_group when you specify a contact?
